sdk/identity/Azure.Identity/tests/ManagedIdentityCredentialArcLiveTests.cs

```diff
@@ -0,0 +1,83 @@
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Threading.Tasks;
+using Azure.Core;
+using Azure.Core.TestFramework;
+using Azure.Security.KeyVault.Secrets;
+using NUnit.Framework;
+
+namespace Azure.Identity.Tests
+{
+    public class ManagedIdentityCredentialArcLiveTests : ManagedIdentityCredentialLiveTestBase
+    {
+        public ManagedIdentityCredentialArcLiveTests(bool isAsync) : base(isAsync)
+        {
+            Matcher = new ArcMiRecordedTestMatcher();
+        }
+
+        [NonParallelizable]
+        [Test]
+        public async Task ValidateSystemAssignedIdentity()
+        {
+            if (string.IsNullOrEmpty(TestEnvironment.ArcEnable))
+            {
+                Assert.Ignore();
+            }
+
+            using (ReadOrRestoreManagedIdentityEnvironment())
+            {
+                var vaultUri = new Uri(TestEnvironment.SystemAssignedVault);
+
+                var cred = new ManagedIdentityCredential(options: InstrumentClientOptions(new TokenCredentialOptions()));
+
+                // Hard code service version or recorded tests will fail: https://github.com/Azure/azure-sdk-for-net/issues/10432
+                var kvoptions = InstrumentClientOptions(new SecretClientOptions(SecretClientOptions.ServiceVersion.V7_0));
+
+                var kvclient = new SecretClient(vaultUri, cred, kvoptions);
+
+                KeyVaultSecret secret = await kvclient.SetSecretAsync("identitytestsecret", "value");
+
+                Assert.IsNotNull(secret);
+            }
+        }
+
+
+        [NonParallelizable]
+        [Test]
+        public void ValidateUserAssignedIdentity()
+        {
+            if (string.IsNullOrEmpty(TestEnvironment.ArcEnable))
+            {
+                Assert.Ignore();
+            }
+
+            using (ReadOrRestoreManagedIdentityEnvironment())
+            {
+                var vaultUri = new Uri(TestEnvironment.SystemAssignedVault);
+
+                var cred = new ManagedIdentityCredential(clientId: Guid.NewGuid().ToString(), options: InstrumentClientOptions(new TokenCredentialOptions()));
+
+                Assert.ThrowsAsync<AuthenticationFailedException>(async () => await cred.GetTokenAsync(new TokenRequestContext(new string[] { AzureAuthorityHosts.GetDefaultScope(AzureAuthorityHosts.AzurePublicCloud) })));
+            }
+        }
+
+        private class ArcMiRecordedTestMatcher : RecordMatcher
+        {
+            public override RecordEntry FindMatch(RecordEntry request, IList<RecordEntry> entries)
+            {
+                var match = base.FindMatch(request, entries);
+
+                if (match.Response.Headers.TryGetValue("WWW-Authenticate", out string[] challenges) && challenges[0].StartsWith("Basic realm="))
+                {
+                    challenges[0] = challenges[0].Split('=')[0] + "=" + Path.Combine(TestContext.CurrentContext.TestDirectory, "Data", "mock-arc-mi-key.key");
```

Comment:
We generally don't use matcher to modify the record. Why is this required?

Reply:
I'm updating the recording in the matcher in this case because the service is responding with a header which contains a path to a file on the local disk. So the idea here is to replace the path the service returned with a static file binplaced when we build the tests. But that path differs depending on the machine your running the tests on, so I can't update it when scrubbing. I originally tried to update the header in the scrubber to be a relative path, but this introduces failures depending on how the tests are run and the working directly.